modules/core/src/lib/attribute.js

```diff
@@ -329,8 +329,8 @@ export default class Attribute extends BaseAttribute {
 
   // PRIVATE HELPER METHODS
   _getVertexOffset(row, bufferLayout) {
+    let offset = this.offset / 4; // Always 4 bytes/element (float, int or uint)
```

Comment:
Shouldn't this be something like `this.offset / this.value.BYTES_PER_ELEMENT`?

Reply:
Unrelated to this PR: `this.offset` is a bit confusing. Maybe `BaseAttribute` could have two properties, `byteOffset` and `elementOffset`?

Reply:
> Shouldn't this be something like `this.offset / this.value.BYTES_PER_ELEMENT`?

Is this true? When we upload colors as `Uint8ClampedArrays`, is element offset `1`?

`this.value.BYTES_PER_ELEMENT` may break because `this.value` can be an array when `constant: true`. A more robust solution would be storing the information as an attribute property.

> Maybe `BaseAttribute` could have two properties, `byteOffset` and `elementOffset`?

I like this idea.

Reply:
> Is this true? When we upload colors as Uint8ClampedArrays, is element offset 1?

I think so, if I'm understanding the `BaseAttribute` API correctly. The offset passed to `vertexAttribPointer` is in bytes, so it would be 1 per element for `Uint8ClampedArray`.